baseframe/forms/validators.py

```diff
@@ -228,11 +197,7 @@ def __init__(self, message=None, timeout=30):
         self.timeout = timeout
 
     def __call__(self, form, field):
-        sniffedmx = self.get_mx(field.data, timeout=self.timeout)
-        if sniffedmx is None or not any([p['public'] for p in sniffedmx['providers']]):
-            # sniffedmx is None only if the domain lookup fails.
-            # This validator will pass in that case because we assume
-            # that most domains are not public email domains.
+        if not is_public_email_domain(field.data, default=False, timeout=self.timeout):
```

Comment:
Are your defaults correct? I get the feeling one of the defaults is wrong.

Reply:
Our assumption for validation is that in case of lookup failure, a domain is not a public email domain. so default is False for both cases. It's `not False` above in `IsNotPublicEmailDomain`, so the validation passes and `False` in `IsPublicEmailDomain`, so validation fails.